datafusion/core/src/execution/options.rs

```diff
@@ -286,7 +314,7 @@ impl<'a> AvroReadOptions<'a> {
 #[derive(Clone)]
 pub struct NdJsonReadOptions<'a> {
     /// The data source schema.
-    pub schema: Option<SchemaRef>,
+    pub schema: Option<&'a Schema>,
```

Comment:
I wonder why you changed this from SchemRef (which is already very fast to copy)?

Reply:
I changed it while trying to standardize the CSV, JSON, and AVRO APIs. I didn't want to break the CSV API so I change others. You are completely right about the problem of the change. Would you suggest changing CSV's schema API (it will break many tests and probably user code.)?

```rust /// Specify schema to use for CSV read
    pub fn schema(mut self, schema: &'a Schema) -> Self {
        self.schema = Some(schema);
        self
    }
```

Reply:
I think changing the CSV API is ok in this case (as it makes the API standard as well as adds new feature support)